### DISCUSSION TITLE
User with different username but same email (type_secret index being unique problem)

### DISCUSSION THREAD

**Discussion by grisinformatica** [codeigniter4/shield#477] [Q&A]
Hello everybody,

As seen here #193 the requirement is that the user has an email, I would add also that for every user the username AND the email has to been unique.

In my particular project, the users identification is based on username/password, but different users can have the same email address for restore password/notifications. Also magic-link could be used if ask for username instead of email.

Don't see a way without modifying the all thing. Thanks in advance.

**Comment by datamweb** [comment]
hello @grisinformatica 
I was really surprised,

> but different users can have the same email address for restore password/notifications. 

Can you explain why this should exist? Can you explain what benefit is created by doing this?

**Reply by grisinformatica** [reply]
Hello @datamweb, well, I quote from README: _shield is designed to be flexible_ and _To cover many auth needs right out of the box, but be simple to add additional functionality to_. I think it is, my comment is not a complaint. 

In my particular approach, users can be minors age. They have usernames or social numbers to identify themselves. They don't have/use email yet, but can create an account with parents email. Also parents or other siblins can create more accounts with the same email.

Shield lets more fields (rather than email or username) be available to be used as credentials for login (phone, social number or id, etc.) But restrict the accounts to a unique email address. These make sense only when using traditional email/password authentication.

Hope I explained myself. Cheers and thanks for reading.

**Reply by datamweb** [reply]
> I quote from README: _shield is designed to be flexible_ and _To cover many auth needs right out of the box...

Yes, this is one of the hallmarks of Shield, I personally love it!

> In my particular approach, users can be minors age...

It was interesting, I think you should code this thing separately, for example, consider a section for parents who can register their children there and give them the necessary access, or something like that.

> Shield lets more fields (rather than email or username) be available ...

Shield currently does not allow you to do this. Now you can use email or username to log in, not any desired field!

**Comment by kenjis** [comment]
This is a requirement that is so far outside of Shield's prerequisites.
So it would be impossible to achieve without extending the code.

First, the email in Shield is the user's own email. The email in this case is, for example, parent_email, 
both of which are emails, but they are different concepts.
